source/MaterialXTest/MaterialXGenShader/GenShaderUtil.cpp

```diff
@@ -659,6 +659,11 @@ void ShaderGeneratorTester::validate(const mx::GenOptions& generateOptions, cons
     copyOptions.skipConflictingElements = true;
     for (const auto& doc : _documents)
     {
+        // For each new file clear the implementation cache.
+        // Since the new file might contain implementations with names
+        // colliding with implementations in previous test cases.
+        context.clearNodeImplementations();
```

Comment:
Does this also have to be done for all places we call generate including rendering unit tests and the varius places in MaterialXView ? 

Or, if the generator has this property/logic, and the MDL generator seems to have added this in explicitly, then  can this explicit step can be removed as it's easy to not do it resulting I assume in unexpected results?

Reply:
This was already done in the render tests but missing here in the generation tests.

But I agree, we could make this safer by always clearing the cache for all generators. For MDL it's obvious that it's needed since we edit the subgraphs there, but it might give issues in all generators. Let's fix this in another change list.